recipe/meta.yaml

```diff
@@ -63,15 +63,537 @@ outputs:
+        - '*/nvcuda.dll'      # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - {{ compiler("cuda") }}                 # [cuda_compiler_version != "None"]
+        # needed for gandiva
+        - clangdev {{ llvm_version }}            # [build_platform != target_platform]
+        - llvmdev {{ llvm_version }}             # [build_platform != target_platform]
+        - gnuconfig                              # [build_platform != target_platform]
+        - cmake
+        - ninja
+        # necessary for vendored jemalloc
+        - autoconf     # [linux]
+        - make         # [linux]
+      host:
+        - {{ pin_subpackage('libarrow', exact=True) }}
+        # for required dependencies, see
+        # https://github.com/apache/arrow/blob/apache-arrow-11.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake#L46-L75
+        - clangdev {{ llvm_version }}
+        - llvmdev {{ llvm_version }}
+        - aws-crt-cpp
+        - aws-sdk-cpp
+        - boost-cpp >=1.70
+        - brotli
+        - bzip2
+        # not yet: https://github.com/conda-forge/cpp-opentelemetry-sdk-feedstock/issues/38
+        # - cpp-opentelemetry-sdk
+        # - proto-opentelemetry-proto =={{ cpp_opentelemetry_sdk }}
+        - gflags
+        - glog
+        - google-cloud-cpp
+        # arrow uses a customized jemalloc, see #944
+        # - jemalloc
+        - libabseil
+        - libgrpc
+        - libprotobuf
+        - libutf8proc
+        - lz4-c
+        - nlohmann_json
+        - orc
+        - rapidjson
+        - re2
+        - snappy
+        - thrift-cpp
+        - ucx          # [linux]
+        - xsimd
+        - zlib
+        - zstd
+        - __cuda >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
+        # since libgoogle-cloud is static on windows, see
+        # https://github.com/conda-forge/google-cloud-cpp-feedstock/pull/108,
+        # its host deps (which aren't yet covered above) leak into the build here
+        - libcrc32c  # [win]
+        - libcurl    # [win]
+      run:
+        - {{ pin_subpackage('libarrow', exact=True) }}
+      run_constrained:
+        - apache-arrow-proc =*={{ build_ext }}
+        # make sure we don't co-install with old version of old package name
+        - arrow-cpp ={{ version }}
+        # old parquet lib output, now part of this feedstock
+        - parquet-cpp <0.0a0
+
+    test:
+      commands:
+        {% set headers = [
+            "arrow/acero/api.h"
+        ] %}
+        {% for each_header in headers %}
+        # headers
+        - test -f $PREFIX/include/{{ each_header }} || (echo "{{ each_header }} not found" && exit 1)  # [unix]
+        - if not exist %LIBRARY_INC%\{{ "\\".join(each_header.split("/")) }} exit 1                    # [win]
+        {% endfor %}
+        {% set libs = (cuda_compiler_version != "None") * ["arrow_cuda"] + [
+            "arrow", "arrow_acero"
+        ] %}
```

Comment:
What's the situation around `libarrow_cuda.so`? It's currently listed as an expected library in a bunch of outputs; if indeed those bits get mashed together in a single library, then these outputs would clobber each other...

Reply:
isn't this just because I am using `libarrow` as a run dependency? I can remove `libarrow` from the run dependency but as `libarrow` also contains all headers at the moment, I should probably stop testing for headers on that case.